tests/other/metadce/mem_no_argv_O3_STANDALONE_WASM_flto.funcs

```diff
@@ -1,4 +1,4 @@
 $_start
-$dlmalloc
 $main
+$malloc
```

Comment:
why did this change?

Reply:
I took a look at this last night.  It seems that its because previously passing `-flto` did *not* link with LTO versions of the system libraries.  One needed to use `WASM_OBJECT_FILES=0` to get that effect.  Now `-flto` is does the correct thing and uses LTO version of the libararies.    Most of our tests used `WASM_OBJECT_FILES=0` on the command line so were not effected.. this test only used `-flto`.   

The name change here has something to do with the way the compiler works when it can see the whole program.